sw/airborne/modules/lidar/tfmini.c

```diff
@@ -155,16 +179,28 @@ static void tfmini_parse(uint8_t byte)
         if (tfmini.distance != 0xFFFF) {
           // compensate AGL measurement for body rotation
           if (tfmini.compensate_rotation) {
-            float phi = stateGetNedToBodyEulers_f()->phi;
             float theta = stateGetNedToBodyEulers_f()->theta;
-            float gain = (float)fabs((double)(cosf(phi) * cosf(theta)));
-            tfmini.distance = tfmini.distance * gain;
+            float ground_distance;
+            if(fabs(theta) < 0.01){
+              ground_distance = 100;  // If it is 0 it is straight
+            }
+            else{
+              ground_distance = lidar_height/sinf(-theta) - lidar_offset;
+            }
+            
+            if ((tfmini.distance >= ground_distance) && (ground_distance > 0)) {
+              tfmini.distance = 0;
+            }
```

Comment:
I still don't get what kind of compensation you are doing here. Can you explain it ? Is it still compatible with AGL measurement from a flying drone ?